### PR TITLE
New GMPE Munson and Thurber 1997 (Volcanic)

### PR DESCRIPTION
**Pull request by klunk386**
This GMPE is needed to run the area source model for Saudi Arabia from Sokolow.
It implements the original GMPE in the main class and a modified version, to empirically account for a different horizontal component combination (Vectorial), in a sub-class.

### CODE REVIEW

**Comment by klunk386** [comment]
Do you think it is useful to include the usual assert for the standard deviation check even if it is only TOTAL?

**Comment by mmpagani** [comment]
Please update the changelog here https://github.com/gem/oq-engine/blob/master/debian/changelog

**Comment by klunk386** [comment]
It seems there is a separate changelog just for hazardlib (changelog_hazardlib). Should I modify this one instead?

**Comment by daniviga** [comment]
No `changelog_hazardlib` is deprecated and kept onlt for historical purposes. `debian/changelog` is the right one.

File: openquake/hazardlib/gsim/munson_thurber_1997.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2013-2017 GEM Foundation
```
Comment (daniviga) [inline openquake/hazardlib/gsim/munson_thurber_1997.py]: If this file is totally new copyright year should be just 2017

File: openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2012-2017 GEM Foundation
```
Comment (daniviga) [inline openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py]: Same as above

File: openquake/hazardlib/gsim/munson_thurber_1997.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+"""
+Module exports :class:`MunsonThurber1997`
+               :class:`MunsonThurber1997Vector`.
+"""
+from __future__ import division
+
+import numpy as np
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
```
Comment (daniviga) [inline openquake/hazardlib/gsim/munson_thurber_1997.py]: `CoeffsTable` is imported but unused

File: openquake/hazardlib/gsim/munson_thurber_1997.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    """
+
+    #: Supported intensity measure component is geometric mean of horizontal
+    #: :attr:`~openquake.hazardlib.const.IMC.VECTORIAL`,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.VECTORIAL
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+
+        base = super(MunsonThurber1997Vector, self)
+        mean, stddevs = base.get_mean_and_stddevs(sites, rup, dists, imt, stddev_types)
```
Comment (daniviga) [inline openquake/hazardlib/gsim/munson_thurber_1997.py]: Line is too long

File: openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+from openquake.hazardlib.gsim.munson_thurber_1997 import MunsonThurber1997
+
+from openquake.hazardlib.tests.gsim.utils import BaseGSIMTestCase
+
+
+class MunsonThurber1997TestCase(BaseGSIMTestCase):
+    GSIM_CLASS = MunsonThurber1997
+
```
Comment (daniviga) [inline openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py]: Whitespaces

File: openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+class MunsonThurber1997TestCase(BaseGSIMTestCase):
+    GSIM_CLASS = MunsonThurber1997
+    
+    # Test data were taken from Table 5 of Munson and Thurber (1997)
+    # Standard deviation is a fixed value
+
+    def test_mean_normal(self):
+        self.check('MT97/MT97_MEAN.csv',
+                    max_discrep_percentage=0.6)
```
Comment (daniviga) [inline openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py]: over indented

File: openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_std_total(self):
+        self.check('MT97/MT97_STD_TOTAL.csv',
+                    max_discrep_percentage=0.1)
```
Comment (daniviga) [inline openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py]: over indented

Comment (mmpagani) [inline openquake/hazardlib/tests/gsim/munson_thurber_1997_test.py]: You must add the test on the TOTAL standard deviation. At present when you call `mean, stddev = gsim.get_mean_and_stddevs(sctx, rctx, dctx, imt, [const.StdDev.INTER_EVENT])` in stddev you get the values of the TOTAL std which is clearly wrong.

**Comment by mmpagani** [comment]
You must add the test on the TOTAL standard deviation. At present when you call mean, stddev = gsim.get_mean_and_stddevs(sctx, rctx, dctx, imt, [const.StdDev.INTER_EVENT]) in stddev you get the values of the TOTAL std which is clearly wrong.

**Comment by daniviga** [comment]
https://github.com/gem/oq-engine/pull/2973 has been merged. Please sync with master and make sure travis is executing hazardlib tests.

**Comment by daniviga** [comment]
Travis is unhappy: please add the new GSIM to `https://github.com/gem/oq-engine/blob/master/doc/sphinx/openquake.hazardlib.gsim.rst`

**Comment by mmpagani** [comment]
LGTM

**Pull request merged**